pyiron_workflow/composite.py

```diff
@@ -637,28 +637,46 @@ def _child_signal_connections(
 
     def __getstate__(self):
         state = super().__getstate__()
+        # Store connections as strings
         state["_child_data_connections"] = self._child_data_connections
         state["_child_signal_connections"] = self._child_signal_connections
-        # Bidict implements a custom reconstructor that is not playing well with h5io
+
+        # Transform the IO maps into a datatype that plays well with h5io
+        # (Bidict implements a custom reconstructor, which hurts us)
         state["_inputs_map"] = (
             None if self._inputs_map is None else dict(self._inputs_map)
         )
         state["_outputs_map"] = (
             None if self._outputs_map is None else dict(self._outputs_map)
         )
+
+        # Remove the nodes container from the state and store each element (node) right
+        # in the state -- the labels are guaranteed to not be attributes already so
+        # this is safe, and it makes sure that the storage path matches the graph path
+        del state["nodes"]
+        state["node_labels"] = list(self.nodes.keys())
+        for node in self:
+            state[node.label] = node
```

Comment:
Ah thanks for the comments XD

Reply:
This is actually fine since child node labels are already [guaranteed to not conflict with anything in the parent's `__dir__`](https://github.com/pyiron/pyiron_workflow/blob/f7e810c86a049f555836b41ddfc8d33878b6b15e/pyiron_workflow/composite.py#L335).

However, this was an excellent point since (a) that promise was not yet articulated in the `Composite` docstring, (b) it's a good idea to add an explanatory comment here for future devs (I don't want to re-execute the test for performance reasons, although the cost would be small), and (c) the `state["node_labels"]` item stored right above was not similarly safe since `node_labels` was not a `__dir__` item yet(!!!)

I patched these in three commits.